clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -453,9 +453,16 @@ private TachyonByteBuffer readLocalByteBuffer(int blockIndex, long offset, long
    */
   TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) {
     // We call into the remote block in stream class to read a remote byte buffer
-    ByteBuffer buf =
-        RemoteBlockInStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length,
-            mTachyonConf);
+    // TODO: Avoid using this constructor, investigate the right way to read remote buffer here.
+    RemoteBlockInStream inStream = new RemoteBlockInStream(new TachyonFile(null, -1, null),
+        ReadType.NO_CACHE, -1, null, null, true);
+    ByteBuffer buf
+        = inStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length, mTachyonConf);
```

Comment:
Yes. Is there any copy involved right now? If not, then there probably should be, right? Any client/user can call this readRemoteByteBuffer(), and use the result for as long as they want, but the RemoteBlockInStream should be closed() to possibly release the internal buffer.